posthog/temporal/usage_reports/run_usage_reports.py

```diff
@@ -0,0 +1,197 @@
+
+            # First capture the events to PostHog
+            if not inputs.skip_capture_event:
+                try:
+                    at_date_str = at_date.isoformat() if at_date else None
+                    capture_report(
+                        pha_client=pha_client,
+                        organization_id=organization_id,
+                        full_report_dict=full_report_dict,
+                        at_date=at_date_str,
+                    )
+                except Exception as err:
+                    logger.exception(f"Error capturing report for organization {organization_id}: {err}")
+
+            # Then send the reports to billing through SQS (only if the producer is available)
+            if has_non_zero_usage(full_report) and producer:
+                try:
+                    _queue_report(producer, organization_id, full_report_dict)
+                    total_orgs_sent += 1
+                except Exception as err:
+                    logger.exception(f"Error queueing report for organization {organization_id}: {err}")
+        except Exception as loop_err:
+            logger.exception(f"Error processing organization report: {loop_err}")
+
+    pha_client.capture(
+        "internal_billing_events",
+        "organization usage report complete",
+        {
+            "total_orgs": total_orgs,
+            "period_start": period_start.isoformat(),
+            "period_end": period_end.isoformat(),
+            "total_orgs_sent": total_orgs_sent,
+            "region": get_instance_region(),
+        },
+        groups={"instance": settings.SITE_URL},
+    )
+
+
+@workflow.defn(name="run-usage-reports")
+class RunUsageReportsWorkflow(PostHogWorkflow):
+    @staticmethod
+    def parse_inputs(inputs: list[str]) -> RunUsageReportsInputs:
+        """Parse inputs from the management command CLI."""
+        loaded = json.loads(inputs[0])
+        return RunUsageReportsInputs(**loaded)
+
+    @workflow.run
+    async def run(self, inputs: RunUsageReportsInputs) -> str:
+        query_usage_reports_inputs = QueryUsageReportsInputs(
+            at=inputs.at,
+        )
+        query_usage_reports_result = await workflow.execute_activity(
+            query_usage_reports,
+            query_usage_reports_inputs,
+            start_to_close_timeout=timedelta(minutes=20),
+        )
+
+        if not query_usage_reports_result.org_reports:
+            return "success - no reports to send"
+
+        send_usage_reports_inputs = SendUsageReportsInputs(
+            org_reports=query_usage_reports_result.org_reports,
+            period=query_usage_reports_result.period,
+            skip_capture_event=inputs.skip_capture_event,
+            at=inputs.at,
+        )
+        await workflow.execute_activity(
+            send_usage_reports,
+            send_usage_reports_inputs,
+            start_to_close_timeout=timedelta(minutes=10),
+        )
+
+        return "success"
```

Comment:
It's usually a good idea to wrap the workflow in a try/catch incase there's some unexpected error that you want logged out